frontend/device/remarkable/device.lua

```diff
@@ -60,18 +78,35 @@ local adjustTouchEvt = function(self, ev)
     end
 end
 
+
 function Remarkable:init()
     self.screen = require("ffi/framebuffer_mxcfb"):new{device = self, debug = logger.dbg}
-    self.powerd = require("device/remarkable/powerd"):new{device = self}
+    self.powerd = require("device/remarkable/powerd"):new{
+        device = self,
+        capacity_file = self.battery_path,
+        status_file = self.status_path,
+    }
     self.input = require("device/input"):new{
         device = self,
         event_map = require("device/remarkable/event_map"),
     }
 
-    self.input.open("/dev/input/event0") -- Wacom
-    self.input.open("/dev/input/event1") -- Touchscreen
-    self.input.open("/dev/input/event2") -- Buttons
-    self.input:registerEventAdjustHook(adjustTouchEvt)
+    self.input.open(self.input_wacom) -- Wacom
+    self.input.open(self.input_ts) -- Touchscreen
+    self.input.open(self.input_buttons) -- Buttons
+
+    self.input:registerEventAdjustHook(adjustAbsEvt)
+
+    if self.invertX() then
+        self.input:registerEventAdjustHook(self.input.adjustTouchMirrorX, self.mt_width)
+    end
+
+    local mt_scale_x = self.mt_width / screen_width
+    local mt_scale_y = self.mt_height / screen_height
+
+    self.input:registerEventAdjustHook(self.input.adjustTouchMirrorY, self.mt_height)
+    self.input:registerEventAdjustHook(self.input.adjustTouchScale, {x=mt_scale_x, y=mt_scale_y})
```

Comment:
this screws the pen scaling

Reply:
@ddvk https://github.com/ddvk/koreader/pull/2 is a quck workaround.